### PR TITLE
Circles 336 gened endpoint

### PR DESCRIPTION
**Pull request by sonar3d**

### CODE REVIEW

**Review by joooooooooooooooooooooooooooooooooooosh — APPROVED** [review]

File: backend/server/routers/programs.py [hunk 1]
```diff
@@ -211,4 +212,21 @@ def get_structure(
...
+        200: {
+            "description": "Returns all geneds available to a given to the given code",
+        },
+    },
+)
+
+def getGenEds(
+    programCode: str, search = None):
+    all_geneds = data_helpers.read_data("data/scrapers/genedPureRaw.json")
+    return {"general education courses" : all_geneds[programCode]}
```
Comment (gtangelo) [inline backend/server/routers/programs.py]: Im assuming this is the key if the payload for this endpoint? Really nitpick but can we name this to 'courses' instead? (if that's better? Haha) 😅

File: backend/server/routers/programs.py [hunk 2]
```diff
@@ -211,4 +212,21 @@ def get_structure(
...
+
+@router.get(
+    "/getGenEds/{programCode}",
+    responses={
+        400: {
+            "description": "The given program code could not be found in the database",
+        },
+        200: {
+            "description": "Returns all geneds available to a given to the given code",
+        },
```
Comment (Peedee2002) [inline backend/server/routers/programs.py:245]: Would you mind giving a more concrete example here?

File: backend/server/routers/programs.py [hunk 3]
```diff
@@ -211,4 +212,21 @@ def get_structure(
...
+    },
+)
+
+def getGenEds(
+    programCode: str, search = None):
```
Comment (Peedee2002) [inline backend/server/routers/programs.py]: Search doesn't seem to do anything? Is that intended?

File: backend/server/routers/programs.py [hunk 4]
```diff
@@ -211,4 +212,21 @@ def get_structure(
...
 
     
-    return {"structure": structure}
\ No newline at end of file
+    return {"structure": structure}
+
+@router.get(
+    "/getGenEds/{programCode}",
+    responses={
+        400: {
```
Comment (Peedee2002) [inline backend/server/routers/programs.py:222]: There should be a response model here. You can use Courses if you take Gabriel's advice.

File: backend/server/routers/programs.py (hunk 2 above)
Comment (sonar3d) [reply backend/server/routers/programs.py:245]: ah yes, only didn't cause the course list is so long

File: backend/server/routers/programs.py [hunk 5]
```diff
@@ -211,4 +213,472 @@ def get_structure(
...
+                            "COMP1521",
+                            "COMP1531",
+                            "COMP1911",
+                            "COMP2511",
+                            "COMP2521",
+                            "COMP3511",
+                            "COMP6441",
+                            "COMP6443",
+                            "COMP6445",
+                            "COMP6447",
```
Comment (Peedee2002) [inline backend/server/routers/programs.py]: Ok shit lol that's massive - have a few as an example and cut it off lol

**Review by Peedee2002 — APPROVED** [review]
Just that one thing and we should be good

**Review by Peedee2002 — APPROVED** [review]
Lgtm

**Pull request merged**